lib/JSON/Validator.pm

```diff
@@ -862,6 +862,10 @@ sub _validate_type_object {
     return E $path, _expected(object => $data);
   }
 
+  if (defined $schema->{maxSize} && length(Mojo::JSON::encode_json($data)) > $schema->{maxSize}) {
```

Comment:
Is it possible to fail at serializing the `$data` here or we have validate that before we reach this point?

Reply:
I did some additional testing on this. $data here is always a perl object and can be serialized. If it's not an object it will already have failed before this point.